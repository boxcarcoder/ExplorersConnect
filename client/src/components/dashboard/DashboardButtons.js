import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

const DashboardButtons = () => {
  return (
    <Fragment>
      <div className='buttons vert-m-1'>
        <Link to='/edit-profile' className='btn'>
          <i className='fas fa-user-circle text-primary'></i> Edit profile
        </Link>
        <Link to='/add-destination' className='btn'>
          <i className='fas fa-map-signs text-primary'></i> Add Destinations
        </Link>
        <Link to='/add-gear' className='btn'>
          <i className='fas fa-snowboarding text-primary'></i> Add Gear
        </Link>
      </div>
    </Fragment>
  );
};

export default DashboardButtons;
