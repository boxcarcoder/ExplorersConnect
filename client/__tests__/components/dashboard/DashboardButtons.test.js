import React from 'react';
import { shallow } from 'enzyme';

// Import the non-connected component. The default export for DashboardButtons is non-connected.
import DashboardButtons from '../../../src/components/dashboard/DashboardButtons';

// Globals for tests
let wrapper;

describe('<DashboardButtons /> component.', () => {
  beforeEach(() => {
    wrapper = shallow(<DashboardButtons />);
  });

  test('Successfully renders.', () => {
    expect(wrapper.exists()).toBe(true);
  });

  describe('Includes a link to', () => {
    test('edit profile.', () => {
      // Search for the link to /edit-profile.
      const link = wrapper.find('Link').at(0);

      // Execute the test.
      expect(link.props().to).toBe('/edit-profile');
    });

    test('add destinations.', () => {
      // Search for the link to /edit-profile.
      const link = wrapper.find('Link').at(1);

      // Execute the test.
      expect(link.props().to).toBe('/add-destinations');
    });

    test('add gears.', () => {
      // Search for the link to /edit-profile.
      const link = wrapper.find('Link').at(2);

      // Execute the test.
      expect(link.props().to).toBe('/add-gears');
    });
  });
});
