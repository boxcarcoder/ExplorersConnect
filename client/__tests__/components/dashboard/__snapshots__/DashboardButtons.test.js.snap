// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardButtons /> component. Successfully renders. 1`] = `
<Fragment>
  <div
    className="buttons vert-m-1"
  >
    <Link
      className="btn"
      to="/edit-profile"
    >
      <i
        className="fas fa-user-circle text-primary"
      />
       Edit profile
    </Link>
    <Link
      className="btn"
      to="/add-destinations"
    >
      <i
        className="fas fa-map-signs text-primary"
      />
       Add Destinations
    </Link>
    <Link
      className="btn"
      to="/add-gears"
    >
      <i
        className="fas fa-snowboarding text-primary"
      />
       Add Gear
    </Link>
  </div>
</Fragment>
`;
